src/binder/bind/read/bind_in_query_call.cpp

```diff
@@ -21,28 +22,38 @@ std::unique_ptr<BoundReadingClause> Binder::bindInQueryCall(const ReadingClause&
     auto expr = call.getFunctionExpression();
     auto functionExpr = expr->constPtrCast<ParsedFunctionExpression>();
     auto functionName = functionExpr->getFunctionName();
-    expression_vector params;
+    expression_vector children;
     for (auto i = 0u; i < functionExpr->getNumChildren(); i++) {
-        auto child = functionExpr->getChild(i);
-        params.push_back(expressionBinder.bindExpression(*child));
+        auto childExpr = functionExpr->getChild(i);
+        children.push_back(expressionBinder.bindExpression(*childExpr));
     }
     TableFunction tableFunction;
-    std::vector<Value> inputValues;
     std::vector<LogicalType> inputTypes;
-    for (auto& param : params) {
-        ExpressionUtil::validateExpressionType(*param, ExpressionType::LITERAL);
-        auto literalExpr = param->constPtrCast<LiteralExpression>();
+    for (auto& child : children) {
+        ExpressionUtil::validateExpressionType(*child, ExpressionType::LITERAL);
+        auto literalExpr = child->constPtrCast<LiteralExpression>();
         inputTypes.push_back(literalExpr->getDataType());
-        inputValues.push_back(*literalExpr->getValue());
     }
     auto catalogSet = clientContext->getCatalog()->getFunctions(clientContext->getTx());
     auto functionEntry = BuiltInFunctionsUtils::getFunctionCatalogEntry(functionName, catalogSet);
     if (functionEntry->getType() != CatalogEntryType::TABLE_FUNCTION_ENTRY) {
         throw BinderException(stringFormat("{} is not a table function.", functionName));
     }
-    auto func = BuiltInFunctionsUtils::matchFunction(functionExpr->getFunctionName(), inputTypes,
-        functionEntry);
+    auto func = BuiltInFunctionsUtils::matchFunction(functionName, inputTypes, functionEntry);
     tableFunction = *func->constPtrCast<TableFunction>();
+    std::vector<Value> inputValues;
```

Comment:
> Plus, if were to implement casting. I believe the best practice it to pass expression instead of value because there is no guarantee that we can always fold an expression at compile time.
I tried it but for in query call functions, we need to know the value of the literal expression before we run ```bindFunc```, since we'll use this string as ```tableName``` to find ```tableEntry``` for ```bindData```.